### PR TITLE
fix: verify the local branch is up to date with the remote one

### LINKED ISSUE #709: release fails because "git push --dry-run" fails with  "HEAD -> master (non-fast-forward)"
## Current behavior

`semantic-release` is checking if it has write access with 

```
git push --dry-run https://[secure]@github.com/octokit/routes.git HEAD:master
```

It currently fails the release if the server rejects because it’s fast forward

```
! [rejected]        HEAD -> master (non-fast-forward)
error: failed to push some refs to 'https://[secure]@github.com/octokit/routes.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
````

## Expected behavior

I think if we can take this error as a success, as we wouldn’t get it if we wouldn’t have write access, I assume?

## Environment

- **semantic-release** version: 15.1.2
- CI environment: Travis
- CI logs: https://travis-ci.org/octokit/routes/jobs/355627978#L2104

### PR DESCRIPTION
**Pull request by pvdlg**
Fix #709

### CODE REVIEW

File: lib/git.js [hunk 1]
```diff
@@ -129,6 +129,17 @@ async function verifyTagName(tagName) {
   }
 }
 
+/**
+ * Verify the local branch is up to date with the remote one.
+ *
+ * @param {String} branch The repositoru branch for which to verify status.
```
Comment (gr2m) [inline lib/git.js]: /s/repositoru/repository

File: test/helpers/git-utils.js [hunk 2]
```diff
@@ -212,6 +209,15 @@ export async function gitCommitTag(gitHead) {
  * @param {String} branch The branch to push.
  * @throws {Error} if the push failed.
  */
-export async function push(origin, branch) {
+export async function push(origin = 'origin', branch = 'master') {
```
Comment (gr2m) [inline test/helpers/git-utils.js]: can you please explain why you set defaults now? Could this be  problem if `master` is not the default branch on remote?

And unrelated to this PR: should the `origin` argument be called `remote`? `"origin"` is the default remote name, or does the parameter refer to something else here?

Comment (pvdlg) [reply test/helpers/git-utils.js]: This code is not used in semantic-release. It's just code used in the test.
I set the defaults out of convenience.

Yes I can change the arg name to `remote`

Comment (pvdlg) [reply test/helpers/git-utils.js]: Well actually in each other function on this util file similar args are named `origin`. Should I change it to `remote` everywhere? Or maybe `repositoryUrl` would be clearer?

Comment (gr2m) [reply test/helpers/git-utils.js]: `repositoryUrl` is more clear I think 👍

**Pull request merged**